docs/developer/payments/price-calculations.mdx

```diff
@@ -0,0 +1,179 @@
+---
+title: Price calculations
+---
+
+This document provides an overview of how various transaction amount fields in a `TransactionItem` are calculated and influenced. Additionally, it provides details on how `totalBalance` is calculated for Checkout and Order entities.
+
+## Transaction Amount Fields Overview
+
+The following fields are calculated based on transaction events and their types:
+
+- **`authorizedAmount`**: The total amount successfully authorized.
+- **`chargedAmount`**: The total amount successfully charged.
+- **`refundedAmount`**: The total amount successfully refunded.
+- **`canceledAmount`**: The total amount successfully canceled.
+- **`authorizePendingAmount`**: The total amount of pending authorizations.
+- **`chargePendingAmount`**: The total amount of pending charges.
+- **`refundPendingAmount`**: The total amount of pending refunds.
+- **`cancelPendingAmount`**: The total amount of pending cancellations.
+
+### Calculation process
+
+When the transaction amounts are recalculated, firstly all values are set to zero, 
+then the calculations are applied in the following order:
+  1. recalculate amounts based on events without PSP reference
+  2. recalculate amounts based on `AUTHORIZATION` events
+  3. recalculate amounts based on `CHARGE` events
+  4. recalculate amounts based on `REFUND` events
+  5. recalculate amounts based on `CANCEL` events
+
+### Assumptions
+
+#### 1. Event Grouping:
+  - Events are grouped based on their PSP reference and type of action (e.g., authorization, charge).
+  - Grouping ensures that events of the same type and PSP reference are properly matched to calculate amounts accurately.
+#### 2. Pending Amounts:
+  - Pending amounts (`authorizePendingAmount`, `chargePendingAmount`, `refundPendingAmount`, `cancelPendingAmount`) are increased only if a `REQUEST` event exists for the corresponding PSP reference.
+  - If a success or failure event is also associated with the same PSP reference, the system assumes the requested amount has already been processed, and the pending amount will not be increased.
+#### 3. `SUCCESS` and `FAILURE` Events:
```

Comment:
Wouldn't be also good to mention that in case of the SUCCESS, the amount will be subtracted from the "previous" value? 
I mean, we have a transaction with event `AUTHORIZE_SUCCESS` and amount 100. TR has authorizedAmount:100, chargedAmount:0. Then we receive the event CHARGE_SUCESS and amount: 100. The 100 will have an impact on authorizedAmount. TR will have: authorizedAmount:0, chargedAmount:100 (as we subtracted the amount from CHARGE_SUCCESS from authorizedAmount and added it to chargedAmount)

Reply:
Okay, I see that you deeply explained it below. So the question is: Do you think, that this is wort also to mention it here?